### DISCUSSION TITLE
table: how to add border to heading only?

### DISCUSSION THREAD

**Discussion by bashbunni** [charmbracelet/lipgloss#246] [Q&A] [answered]
**Describe the bug**

Desired output: (but I'm going to make heading pink instead of first element)
![goal](https://github.com/charmbracelet/lipgloss/assets/15822994/5039fe90-b62e-4048-af90-d18e1cd7e901)

I'm trying to add a bottom border so only the header is underlined and the rest of the table has hidden borders. The issue is that if I try to add borders to the header through the `StyleFunc`, then I end up with garbled output. 

![garbled output](https://github.com/charmbracelet/lipgloss/assets/15822994/cc33f597-7aeb-4ff8-aad7-cb7acd1753d8)

If I remove all borders but the bottom one, e.g.

```go
			if row == 0 {
				return lipgloss.NewStyle().
					Foreground(lipgloss.Color("212")).
					Border(lipgloss.NormalBorder()).
					BorderTop(false).
					BorderLeft(false).
					BorderRight(false).
					BorderBottom(true).
					Bold(true)
			}

```

then it still doesn't render the bottom-most border

![bottom border only](https://github.com/charmbracelet/lipgloss/assets/15822994/799afb64-06fb-4c52-97ca-ebe601848cc8)

**Setup**
Please complete the following information along with version numbers, if applicable.
 - OS [e.g. Ubuntu, macOS]: macOS
 - Shell [e.g. zsh, fish]: zsh
 - Terminal Emulator [e.g. kitty, iterm]: kitty
 - Terminal Multiplexer [e.g. tmux]: N/A
 - Locale [e.g. en_US.UTF-8, zh_CN.UTF-8, etc.]: en_US.UTF-8

**To Reproduce**
1. open up the repo (linked)
2. add a task: `go run . add "pet dog" -p "pet care"`
3. run the program with `go run . list`

**Source Code**
Please include source code if needed to reproduce the behavior. 

https://github.com/charmbracelet/taskcli/blob/feat/lipgloss-table/cmds.go#L124

**Expected behavior**
I would expect to be able to apply my own borders to the heading element.

**Screenshots**
Add screenshots to help explain your problem.

**Additional context**

**Comment by maaslalani** [comment, marked as answer]
Hey! So the `StyleFunc` is used to style each cell which is why when you apply a border to the cell it tries to add a border to each cell and gives you that weird result. We should probably handle this a little bit better.

The `BorderHeader` is applied with the Border supplied on the border so when given `lipgloss.HiddenBorder` the `BorderHeader` also applies with a hidden border. This is so that it is consistent with the rest of the table.

I think this can be solved through a custom border, would this style be good?

<img width="416" alt="image" src="https://github.com/charmbracelet/lipgloss/assets/42545625/2f36ecc8-2533-4532-ab72-e16b360aba3c">

If so, the following custom border can be used:

```go
Border(lipgloss.Border{
	Top:          "─",
	Bottom:       " ",
	Left:         " ",
	Right:        " ",
	TopLeft:      "─",
	TopRight:     "─",
	BottomLeft:   " ",
	BottomRight:  " ",
	MiddleLeft:   "─",
	MiddleRight:  "─",
	Middle:       "─",
	MiddleTop:    "─",
	MiddleBottom: " ",
}).BorderHeader(true).
```

**Comment by maaslalani** [comment]
The more correct solution here would be to allow a custom border to be passed to use on the header itself, or to allow fully custom styling of the border header but I suspect this is a niche use case. What do you think?

**Reply by bashbunni** [reply]
Yeah I think it's pretty niche. I don't think we need to add anything to accommodate this, I thought I had maybe just overlooked something. Thanks for clarifying :)
